src/pixelator/graph/backends/implementations/_networkx.py

```diff
@@ -693,3 +705,69 @@ def subgraphs(self) -> Iterable[Graph]:
             Graph(NetworkXGraphBackend(self._graph.subgraph(cluster).copy()))
             for cluster in self._clustering
         ]
+
+
+def pmds_layout(
+    g: nx.Graph, pivots: int = 50, dim: int = 2, seed: Optional[int] = None
+) -> pd.DataFrame:
+    """Add some number of vertices to the graph instance.
+
+    :param g: A networkx graph object
+    :param pivots: The number of pivot nodes to use
+    :param dim: The dimension of the layout
+    :param seed: Set seed for reproducibility
+
+    :return: A dataframe with layout coordinates
+    :rtype: pd.DataFrame
+    :raises: ValueError raises if conditions are not met
+    """
+    if not isinstance(g, nx.Graph):
+        raise ValueError("Not a graph object")
+
+    if not nx.is_connected(g):
+        raise ValueError("Only connected graphs are supported.")
+
+    if pivots >= len(g.nodes):
+        raise ValueError("'pivots' must be less than the number of nodes in the graph.")
+    if pivots < dim:
+        raise ValueError(
+            "'pivots' must be greater than or equal to the dimension of the layout."
+        )
+    if pivots < 0:
```

Comment:
This is not checking for when `pivots` are exactly 0.

Reply:
This check has now been replaced with a check to make sure that pivots >= dim.